.github/workflows/check-performance-regression.yml

```diff
@@ -0,0 +1,53 @@
+#####################################################################################
+# GitHub Action to test performance regression.
+#
+# Workflow starts when:
+# 1) push to master
+# 2) PR created or pushed
+#
+#####################################################################################
+name: Check-Performance-Regression
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches: '*'
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Set up JDK 11
+        uses: actions/setup-java@v4
+        with:
+          java-version: 11
+          distribution: 'temurin'
+
+      - name: Checkout patch
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+          path: ./checkstyle
+
+      - name: Clone JDK 17 Repo
+        uses: actions/checkout@v4
+        with:
+          repository: openjdk/jdk17
+          path: ./checkstyle/.ci/jdk17
```

Comment:
Please make a folder `.ci-temp` on that CI host.

And do  `./checkstyle/.ci-temp/jdk17`

We have a rule to not change repository by CI tasks. This folder already knows by gitignore file

Reply:
done